src/c_api.cpp

```diff
@@ -521,6 +521,24 @@ class Booster {
     return idx;
   }
 
+  size_t GetLargestEvalNameSize() const {
+    size_t max_len = 0;
+    for (const auto& metric : train_metric_) {
```

Comment:
why `train_metric_`?

Reply:
I'm just wrapping LGBM_BoosterGetEvalNames, so I looked into the c_api.cpp and it calls `GetEvalNames` (though it can copy out of bounds), so I just copied the code from `GetEvalNames`:
![image](https://user-images.githubusercontent.com/3976753/76402922-02113c80-637c-11ea-965b-201bc9429c6f.png)

and changed the loop code to figure out the largest name size for the copy.

Is `GetEvalNames` wrong?